controllers/eventhandlers/mapper.go

```diff
@@ -63,49 +66,78 @@ func (r *resourceMapper) EndpointsToService(ctx context.Context, ep *corev1.Endp
 }
 
 func (r *resourceMapper) TargetGroupPolicyToService(ctx context.Context, tgp *v1alpha1.TargetGroupPolicy) *corev1.Service {
-	if tgp == nil {
-		return nil
-	}
-	policyName := k8s.NamespacedName(tgp).String()
+	return policyToTargetRefObj(r, ctx, tgp, &corev1.Service{})
+}
 
-	targetRef := tgp.Spec.TargetRef
-	if targetRef == nil {
-		r.log.Infow("TargetGroupPolicy does not have targetRef, skipping",
-			"policyName", policyName)
-		return nil
-	}
-	if targetRef.Group != coreGroupName || targetRef.Kind != serviceKind {
-		r.log.Infow("Detected non-Service TargetGroupPolicy attachment, skipping",
-			"policyName", policyName, "targetRef", targetRef)
-		return nil
-	}
-	namespace := tgp.Namespace
-	if targetRef.Namespace != nil && namespace != string(*targetRef.Namespace) {
-		r.log.Infow("Detected cross namespace TargetGroupPolicy attachment, skipping",
-			"policyName", policyName, "targetRef", targetRef)
-		return nil
+func (r *resourceMapper) VpcAssociationPolicyToGateway(ctx context.Context, vap *v1alpha1.VpcAssociationPolicy) *gateway_api.Gateway {
+	return policyToTargetRefObj(r, ctx, vap, &gateway_api.Gateway{})
+}
+
+func policyToTargetRefObj[T client.Object](r *resourceMapper, ctx context.Context, policy core.Policy, retObj T) T {
+	null := *new(T)
+	if policy == nil {
+		return null
 	}
+	policyNamespacedName := policy.GetNamespacedName()
 
-	svcName := types.NamespacedName{
-		Namespace: namespace,
+	targetRef := policy.GetTargetRef()
+	if targetRef == nil {
+		r.log.Infow("Policy does not have targetRef, skipping",
+			"policyName", policyNamespacedName)
+		return null
+	}
+	expectedGroup, expectedKind, err := k8sResourceTypeToGroupNameAndKindName(retObj)
+	if err != nil {
+		r.log.Errorw("Failed to get expected GroupVersion of targetRefObj",
```

Comment:
Group, Version, Kind are different concept and this log line is misleading, should be GroupKind or something.